### ISSUE TITLE
Quite difficult to test recent releases on servers without make

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/mail#1395]
So my server (gandi.net) doesn’t have make installed and I don’t have root. I can not quickly git pull and make the app.

Is there any way we can simplify the build stack or does this make no sense? cc @DeepDiver1975 @ChristophWurst @Gomez

**Comment by ChristophWurst** [comment]
I'm still against having dependencies in the git repo. It clutters PRs and you'll get merge/rebase conflicts when rebasing branches (e.g. inside the compiled ``mail.min.js`` and possible with 3rd party libs). Hence, I'm happy with how it is handled now.

What exactly do you mean with "test recent releases"? We have nightly builds for people who want to try the latest master. Additionally we try to release earlier and more often, as discussed a few weeks ago.

If you know how to build the app locally, it's easy to push it to your production server with rsync or similar. I do that regularly if some bug is not reproducible on my local dev env. Would that be a acceptable solution for your situation?

**Comment by ChristophWurst** [comment]
Maybe we can find some kind of bot that creates a build whenever someone pushes to a branch, that would make it even easier.
e.g. Travis could actually do that for us: https://docs.travis-ci.com/user/uploading-artifacts/

**Comment by jancborchardt** [comment]
Yeah, actually an automatically created branch (like the »make« I created) would be cool. I totally agree we should not have dependencies in the repo.

It’s just that for me as a semi-dev it feels a lot more complicated as a whole as well to get the dev setup up and running. Which will prevent people like designers from contributing. So that extra branch would solve that maybe, as well as the setup issue.

**Comment by Gomez** [comment]
The "make" branch is a nice idea. We would use travis to push back with mail.min.js. Here is most of the work done:
https://gist.github.com/vinceallenvince/a7611b10f84e61eebdcc#deploy-to-gh-pages
